### ISSUE TITLE
All public types should be documented

### ISSUE THREAD

**Issue by IngvarKofoed** [neo4j/neo4j-dotnet-driver#40]
When #38 has been resolved, then all public types should be documented to make the driver more user friendly and help developers not make less mistakes and use lesser time on getting starting with the driver.

**Comment by zhenlineo** [comment]
Oh, I replied in #38, all the classes/interfaces that we would like the user to use actually are doc properly (or at least with docs). 
The dotnet API doc could be found [here](http://alpha.neohq.net/docs/dotnet-driver/html/f7d9eea1-5357-f1de-8a54-336a77141b6d.htm) in [alpha channel](http://alpha.neohq.net) 
Also the API doc will also be moved to a public/formal place soon.

Let's revisit this issue when we solved #38.
